### ISSUE TITLE
jshon is not found. Please install this package

### ISSUE THREAD

**Issue by franchino87** [TS3Tools/TS3UpdateScript#75]
Hi there, yesterday i started trying to use this awesome script, sadly i had a Centos6 and i read in this other post of Issue #66 https://github.com/TS3Tools/TS3UpdateScript/issues/66.
So i updated to Centos7 as told in the post above but i'm still unable to find a package for jshon.
####################################
[root@server TS3UpdateScript]# lsb_release -a
LSB Version:    :core-4.1-amd64:core-4.1-noarch:cxx-4.1-amd64:cxx-4.1-noarch:desktop-4.1-amd64:desktop-4.1-noarch:languages-4.1-amd64:languages-4.1-noarch:printing-4.1-amd64:printing-4.1-noarch
Distributor ID: CentOS
Description:    CentOS Linux release 7.4.1708 (Core)
Release:        7.4.1708
Codename:       Core
####################################

I also tried following this link in order to install it manually:
http://kmkeen.com/jshon/
Sadly i don't know what to do with the downloaded Package, i end up with 3 files:
jshon-20120914/Makefile
jshon-20120914/jshon.1
jshon-20120914/jshon.c
After a "make" command i end up with this output and this files:
####################################
[root@server jshon-20120914]# make
cc -std=c99 -Wall -pedantic -Wextra -Werror    -c -o jshon.o jshon.c
cc   jshon.o  -ljansson -o jshon
[root@server jshon-20120914]# ls
Makefile  jshon  jshon.1  jshon.c  jshon.o
####################################
From now on, i really don't know what to do with those files, can someone help me?

This is the output i receive from the script
**--------------------------------------------------------------**
Please wait... Script is working...

Performing self-tests as preventive measures against possible failures[ .. ]

                                                                      [ FAILED ]

jshon is not found. Please install this package
Please set your own administrator eMail in 'configs/config.all'!
One or more self-tests failed. Please fix the above error(s) to continue.

Cleaning up...

Summary of measured execution time for each step
Total script: 4s
Self-Test: 4s
Cleanup: 0s
**--------------------------------------------------------------**

Hope that someone can help.
Thanks

**Comment by Sebi94nbg** [comment]
Thanks for the detailed error description.

Due of that, that the `jshon` package results in more problems, than it helps, I'll try to remove it from the code in order to improve this.

Related issues:
- #66 
- #75

**Comment by franchino87** [comment]
Thanks for your reply, i'll be waiting for the new version so!
Anyway is there some way i can make it work in the meanwhile?

Thanks

**Comment by Sebi94nbg** [comment]
Yes, you can use the parameter `--set-release` and set the release in the `configs/config.all` file: `LATEST_STABLE_RELEASE=3.1.1`

You also need to remove the software check here: https://github.com/TS3Tools/TS3UpdateScript/blob/master/TS3UpdateScript#L2242

Simply removing `jshon` should already work. If not, let me know it, then I'll improve this this evening quickly.

**Comment by franchino87** [comment]
Thanks for your indication, i did the following:
- Changed the string in configs/config.all
- Removed jshon from the check command as you told me to

I then issued the --check --set-release command but sadly i got this error:
###################
-bash: ./ts3server_startscript.sh: Permission denied
1 TeamSpeak 3 server instance found...mission denied
-bash: ./ts3server_startscript.sh: Permission denied
Collecting as much as possible information for following TeamSpeak 3 server instance: /home/teamspeak-server
1 instance for update process...
-bash: ./ts3server_startscript.sh: Permission denied
###################

i guess i got some permission problem (even though looking at your code, you are already doing that check), so i did an ls -l and the permission were:
-rwxrwxrwx 1 root root         2685 Mar  1 00:21 ts3server_startscript.sh

But from now on i guess it is my problem :)
Thanks a lot for your help

**Comment by Sebi94nbg** [comment]
Yeah, the TS3 server files should be owned by a dedicated user and not root.

Just add an extra user like "teamspeak" to your Linux...
`useradd teamspeak`
...and change the ownership of the files to this user...
`chown teamspeak:teamspeak -R /home/teamspeak/teamspeak-server/`
...and restart the server with the correct permissions:
```
# Stop as root user
./ts3server_startscript.sh stop
# Switch to teamspeak user
su - teamspeak
# Start as teamspeak user
./ts3server_startscript.sh start
```

After that, the script should be able to get the status of your server and so on. :)

**Comment by Shoodey** [comment]
Hey I found this to be helpful, it did allow me to install jshon on ubuntu 16
